python/sglang/srt/constrained/fsm_cache.py

```diff
@@ -74,3 +76,10 @@ def fset(self, value):
 
     def init_value(self, regex):
         return RegexGuide(regex, self.outlines_tokenizer)
+
+
+class FSMJsonCache(FSMCache):
```

Comment:
Do not create a new class. Merge them into one with if/else or other better methods.

Reply:
I've changed it for a boolean switch in FSMCache. 
It could be made more modular if BaseToolCache query method could instead take the init_value function as an optional parameter : it would allow for any number of use cases without having to implement a new boolean switch in FSMCache each time. However that doesn't seem necessary at the moment.